pkg/controller/sshd/sshd_controller.go

```diff
@@ -303,44 +295,7 @@ func (r *ReconcileSSHD) Reconcile(request reconcile.Request) (reconcile.Result,
 		}
 	}
 
-	// Create a Route53 DNS entry for the service's load balancer.
-	// TODO Consider using https://github.com/openshift/external-dns
-
-	clusterBaseDomain, err := utils.GetClusterBaseDomain(r.client)
-	if err != nil {
-		r.SetSSHDStatusError(instance, "Failed to get cluster's base domain", err)
-		return reconcile.Result{}, err
-	}
-
-	// Get the ELB-to-be's name from Service's UID.
-	elbName := strings.ReplaceAll("a"+string(foundService.ObjectMeta.UID), "-", "")
-	if len(elbName) > 32 {
-		// truncate to 32 characters
-		elbName = elbName[0:32]
-	}
-
-	exists, elb, err := r.awsClient.DoesELBExist(elbName)
-	if err != nil {
-		r.SetSSHDStatusError(instance, "Failed to get load balancer status", err)
-		return reconcile.Result{}, err
-	}
-	if !exists {
-		// It isn't bad that it doesn't exist if there's no error, so re-queue.
-		r.SetSSHDStatusPending(instance, "Waiting on service load balancers")
-		return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
-	}
```

Comment:
I didn't see any functionality to check if an ELB exists in cloudclient. Is this a gap in the reconcile loop, now?

Reply:
@yithian Refer to https://github.com/openshift/cloud-ingress-operator/blob/3a741b31f9c91bada491f81d512b407d8e148cab/pkg/cloudclient/aws/private.go#L180-L200 in the aws package.

The controller shouldn't need to know anything about the implementation because all it wants is for the DNS entry to be present (or absent) for the associated `Service`. It will call the appropriate interface method to set/delete that record.

However, because it's a common thing that creating the backing cloud load balancer takes time, there is a [bespoke error](https://github.com/openshift/cloud-ingress-operator/blob/3a741b31f9c91bada491f81d512b407d8e148cab/pkg/errors/errors.go#L7-L9) which can be returned by the cloud provider packages to indicate that the ELB (in AWS) isn't created yet. [The controller can make use of that returned error](https://github.com/openshift/cloud-ingress-operator/blob/3a741b31f9c91bada491f81d512b407d8e148cab/pkg/controller/apischeme/apischeme_controller.go#L225-L246) to take some relevant action, such as requeueing in ten seconds.

Reply:
This was super helpful - thanks! I've pushed a commit that should catch that error and requeue.